insonmnia/locator/server.go

```diff
@@ -54,13 +52,39 @@ func (l *Locator) Announce(ctx context.Context, req *pb.AnnounceRequest) (*pb.Em
 		return nil, err
 	}
 
+	var (
+		okIPs      []string
+		skippedIPs []string
+	)
+	for _, strIP := range req.IpAddr {
+		if l.onlyPublicIPs {
+			if ip := net.ParseIP(strIP); ip != nil && !util.IsPrivateIP(ip) {
+				okIPs = append(okIPs, strIP)
+			} else {
+				skippedIPs = append(skippedIPs, strIP)
+			}
+		} else {
+			okIPs = append(okIPs, strIP)
+		}
+	}
+
+	if len(skippedIPs) > 0 {
+		log.G(l.ctx).Info("skipped some announced IPs (only public IPs mode is on)",
+			zap.Stringer("eth", ethAddr),
+			zap.Strings("skipped_ips", skippedIPs))
+	}
+
+	if len(okIPs) < 1 {
+		return nil, errors.New("no white IPs provided")
+	}
+
 	log.G(l.ctx).Info("handling Announce request",
```

Comment:
Please move this line to the beginning of the method. And log all IPs came from the request, it will provide more useful info for debugging. 

But the second logging line with the accepted IPs will be useful too!